platform-api/che-core-api-ssh/src/main/java/org/eclipse/che/api/ssh/server/SshService.java

```diff
@@ -0,0 +1,197 @@
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.eclipse.che.dto.server.DtoFactory.newDto;
+
+/**
+ * Defines Ssh Rest API.
+ *
+ * @author Sergii Leschenko
+ */
+@Api(value = "/ssh", description = "Ssh REST API")
+@Path("/ssh")
+public class SshService {
+    private final SshManager sshManager;
+
+    @Inject
+    public SshService(SshManager sshManager) {
+        this.sshManager = sshManager;
+    }
+
+    /**
+     * Generates and stores ssh pair based on the request
+     *
+     * @param request
+     *         object for setting of new ssh pair
+     * @return stored ssh pair
+     * @throws BadRequestException
+     *         when missed required parameters or parameters is invalid
+     * @throws ServerException
+     *         when some error occurred while generating and saving ssh pair
+     */
+    @POST
+    @Path("generate")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @RolesAllowed("user")
+    //TODO Add swagger annotations for service's methods
+    public SshPairDTO generatePair(GenerateSshPairRequest request) throws BadRequestException, ServerException, ConflictException {
+        requiredNotNull(request, "Generate ssh pair request required");
+        requiredNotNull(request.getService(), "Service name required");
+        requiredNotNull(request.getName(), "Name required");
+
+        return asDto(sshManager.generatePair(getCurrentUserId(), request.getService(), request.getName()));
+    }
+
+    /**
+     * Stores given ssh pair
+     */
+    @POST
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @RolesAllowed("user")
+    public Response addPair(Iterator<FileItem> formData) throws BadRequestException, ServerException, ConflictException {
+        String service = null;
+        String name = null;
+        String privateKey = null;
+        String publicKey = null;
+
+        while (formData.hasNext()) {
+            FileItem item = formData.next();
+            String fieldName = item.getFieldName();
+            switch (fieldName) {
+                case "service":
+                    service = item.getString();
+                    break;
+                case "name":
+                    name = item.getString();
+                    break;
+                case "privateKey":
+                    privateKey = item.getString();
+                    break;
+                case "publicKey":
+                    publicKey = item.getString();
+                    break;
+            }
+        }
+
+        requiredNotNull(service, "Service name required");
+        requiredNotNull(name, "Name required");
+        if (privateKey == null && publicKey == null) {
+            throw new BadRequestException("");//TODO Think about message.
```

Comment:
We should throw exception here if both of private key and public key were missed.